src/views/List.jsx

```diff
@@ -1,18 +1,44 @@
+import { useState, useMemo } from 'react';
+
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [itemSearch, setItemSearch] = useState('');
+
+	const searchedData = useMemo(() => {
+		if (!itemSearch) {
+			return data;
+		} else {
+			return data.filter((item) => {
+				return item.name.toLowerCase().includes(itemSearch);
+			});
+		}
+	}, [data, itemSearch]);
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form>
+				<label htmlFor="itemSearch">Search your shopping list:</label>
+				<input
+					type="text"
+					id="itemSearch"
+					name="itemSearch"
+					value={itemSearch}
+					onChange={(e) => setItemSearch(e.target.value.toLowerCase())}
+				/>
+				{itemSearch ? (
```

Comment:
I really like how the clear button only appears when there's something in the search bar! An alternate way to write this when you don't have something return when `itemSearch` is false would be like this:
```
{itemSearch &&
	<button
		onClick={() => setItemSearch('')}
		aria-label="clear search field"
	>
		x
	</button>
}
```
Here's some docs about it on the old React documentation site! 
https://legacy.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
This makes sense. I updated how the button is rendered.